src/hotspot/share/gc/shared/partialArrayTaskStats.hpp

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+#ifndef SHARE_GC_SHARED_PARTIALARRAYTASKSTATS_HPP
+#define SHARE_GC_SHARED_PARTIALARRAYTASKSTATS_HPP
+
+#include "logging/logHandle.hpp"
+#include "logging/logStream.hpp"
+#include "utilities/globalDefinitions.hpp"
+#include "utilities/growableArray.hpp"
+#include "utilities/macros.hpp"
+
+#if TASKQUEUE_STATS
+
+class outputStream;
+
+// Repository for collecting and reporting statistics about partial array task
+// processing.  Not thread-safe; each processing thread should have its own
+// stats object.
+class PartialArrayTaskStats {
+  size_t _split;
+  size_t _pushed;
+  size_t _stolen;
+  size_t _processed;
+
+  static LogTargetHandle log_target();
+  static bool is_log_enabled();
+  static void print_header(outputStream* s, const char* title);
+  void print_values(outputStream* s, uint id) const;
+  void print_total(outputStream* s) const;
+  void print_values_impl(outputStream* s) const;
+
+  void accumulate(const PartialArrayTaskStats& stats);
+
+public:
+  // All counters are initially zero.
+  PartialArrayTaskStats();
+
+  // Trivially copied and destroyed.
+
+  // Number of arrays split into partial array tasks.
+  size_t split() const { return _split; }
+
+  // Number of partial array tasks pushed onto a queue.
+  size_t pushed() const { return _pushed; }
+
+  // Number of partial array tasks stolen from some other queue.
+  size_t stolen() const { return _stolen; }
+
+  // Number of partial array tasks processed.
+  size_t processed() const { return _processed; }
+
+  void inc_split(size_t n = 1) { _split += n; }
+  void inc_pushed(size_t n = 1) { _pushed += n; }
+  void inc_stolen(size_t n = 1) { _stolen += n; }
+  void inc_processed(size_t n = 1) { _processed += n; }
```

Comment:
I skimmed through callers of these, but can't find a strong reason to use default-arg-value here. Will there be more call-sites that justify this usage?